Created some unit tests to test methods in the Tribler Core package

devos50 opened the pull request:
I've written some small tests to test various methods in the Tribler Core. I also created a separate job (https://jenkins.tribler.org/job/Test_tribler_martijn/) to check the code coverage.

Executing this whole test suite only takes several seconds.

If you have any suggestion for improvements of these tests or their structure, please let me know.

lfdversluis commented on Tribler/Test/Core/test_torrent_utils.py:
```diff
@@ -0,0 +1,31 @@
+        os.remove(result["torrent_file_path"])
+
+    def test_create_torrent_two_files(self):
+        create_torrent_file([self.TORRENT_DATA_DIR + "/file1.txt", self.TORRENT_DATA_DIR + "/file2.txt"],
```
you may want to also move `file1.txt` and `file2.txt` to an instance variable. `file1.txt` is now used at two places, changing only one of them (if you decide to) would break your test :)

devos50 commented:
@whirm I would like to have this one merged before I start writing more tests. I will add new tests in small PRs.

I added a newline to all files and replaced the built-in assert with the assert provided by nose.

whirm commented on Tribler/Test/Core/test_encoding.py:
```diff
@@ -0,0 +1,192 @@
+    _a_encode_bytes, _a_encode_list, _a_decode_true, _a_decode_false, _a_decode_none, _a_encode_set, _a_encode_none, \
+    _a_encode_bool, _a_encode_tuple, _a_encode_dictionary, bytes_to_uint, _a_decode_int, _a_decode_long, _a_decode_float, \
+    _a_decode_unicode, encode, _a_decode_bytes, _a_decode_list, _a_decode_mapping, _a_decode_set, _a_decode_tuple, \
+    _a_decode_dictionary, decode
```
Sorry I didn't see that one before, but we use parens-wrapped multiline imports in Tribler.

whirm commented on Tribler/Test/Core/test_install_dir.py:
```diff
@@ -0,0 +1,8 @@
+class TriblerCoreTestInstallDir(TriblerCoreTest):
+
+    def test_install_dir(self):
+        self.assertIsInstance(determine_install_dir(), str)
```
Maybe we should check that the path returned is the right one? Better to do it now or we will forget about it.

whirm commented on Tribler/Test/Core/test_osutils.py:
```diff
@@ -7,7 +7,8 @@
 
-from Tribler.Core.osutils import fix_filebasename
+from Tribler.Core.osutils import fix_filebasename, is_android, get_home_dir, get_appstate_dir, get_picture_dir, \
+    get_desktop_dir
```
here too

whirm commented on Tribler/Test/Core/test_permid.py:
```diff
@@ -0,0 +1,31 @@
+        loaded_key = permid.read_keypair(key_pair_path)
+        self.assertIsInstance(loaded_key, EC)
+
+        shutil.rmtree(unicode(test_base_dir), ignore_errors=True)
```
missing EOL

whirm replied:
I guess you can configure your editor to add these?

lfdversluis commented on Tribler/Test/Core/test_torrent_def.py:
```diff
@@ -0,0 +1,368 @@
+# Written by Arno Bakker
+# see LICENSE.txt for license information
+#
+# TODO:
```
todo?

devos50 commented:
@lfdversluis @whirm thanks for the feedback, I edited my PR where necessary.

whirm commented:
@devos50 also, for next time, remember to have separated commits when moving and then modifying files, so git can follow the file's history correctly.

devos50 commented:
@whirm ok thanks, will keep that in mind!
